### ISSUE TITLE
json/decode is called twice with clj-http if api operation.response.content is "*/*"

### ISSUE THREAD

**Issue by felixdo** [oliyh/martian#189]
Hi, thanks for this amazing tool. It's what I've been looking for a long time.

I'm consuming an api that has an operation with response.content set to "*/*", whereas it actually produces application/json. 
I'm using clj-http, and for this case, martian uses its { :as :auto } option. The server responds with content-type: application/json, so clj/http auto-coerces the result into a map, whereas the martian coercion interceptor still accepts a string, and then fails when it calls cheshire to decode. I wonder what the best way to mitigate this would be (apart from fixing the api spec).

**Comment by felixdo** [comment]
Yes that would work, but not sure if it's the best solution. Maybe its not needed if we get it right earlier.

As a workaround, I just remove `:as :auto`  in another interceptor that I insert after interceptors/coerce-response. 

Maybe one could allow a customizable auto-encoder (e.g. via a special key in the encoders map)
I'm also not sure if there's really a valid use case to ever pass `:as :auto`, I would maybe consider removing it.
Then also.. you dont want to break existing clients, so maybe customizable auto-encoder is the way to go.

**Comment by marksto** [comment]
Removing `:as :auto` makes total sense, since the Babashka HTTP client doesn't support this option. I had issues with it during the #230 development (https://github.com/oliyh/martian/pull/230/commits/ab49e72b91bc722e89dffda0ae5e4e748bc22834). I think this also relates to #58.

**Comment by marksto** [comment]
Hi @felixdo! This should be fixed now. You should be able to test it after the upcoming major release.
